service/controller/v15/resource/cloudformation/create.go

```diff
@@ -29,18 +28,6 @@ func (r *Resource) ApplyCreateChange(ctx context.Context, obj, createChange inte
 			return microerror.Mask(err)
 		}
 
```

Comment:
All that hackery removed from `cloudformation` resource.

Reply:
I think it was a hack from the very beggining. Hard to say why it was designed like that.
I don't have ideas how to raise awareness of those things. We should just keep in mind that all we write should be idempotent. But that should be actually the company mindset.
I hope logging guides we have now will make it more visible and clear. I'm happy see more guidelines, but I don't think we can describe every case. At some point we just have to rely on each other.